egs2/librispeech/hubert_asr1/hubert_asr.sh

```diff
@@ -0,0 +1,552 @@
+		_opts+="--train_shape_file ${_split_dir}/text_shape.word "
+		_opts+="--multiple_iterator true "
+		
+	    else
+		_opts+="--train_data_path_and_name_and_type ${_asr_train_dir}/${_scp},speech,${_type} "
+		_opts+="--train_data_path_and_name_and_type ${_asr_train_dir}/text,text,text "
+		_opts+="--train_shape_file ${asr_stats_dir}/train/speech_shape "
+		_opts+="--train_shape_file ${asr_stats_dir}/train/text_shape.word "
+	    fi
+
+	    if [ -z "${asr_tag}" ]; then
+		if [ -n "${asr_config}" ]; then
+		    asr_tag="$(basename "${asr_config}" .yaml)_${feats_type}"
+		else
+		    asr_tag="train_${feats_type}"
+		fi
+	    fi
+	    
+	    if [ -z "${asr_exp}" ]; then
+		asr_exp="${expdir}/pretrain_${asr_tag}_iter${iter}"
+	    fi
+
+	    log "Generate '${asr_exp}/run.sh'. You can resume the process from stage 5 using this script"
+	    mkdir -p "${asr_exp}"; echo "${run_args} --stage 5 \"\$@\"; exit \$?" > "${asr_exp}/run.sh"; chmod +x "${asr_exp}/run.sh"
+	    
+	    # NOTE(kamo): --fold_length is used only if --batch_type=folded and it's ignored in the other case
+	    log "ASR training started... log: '${asr_exp}/train.log'"
+	    if echo "${cuda_cmd}" | grep -e queue.pl -e queue-freegpu.pl &> /dev/null; then
+		# SGE can't include "/" in a job name
+		jobname="$(basename ${asr_exp})"
+	    else
+		jobname="${asr_exp}/train.log"
+	    fi
+	    
+	    # shellcheck disable=SC2086
+	    ${python} -m espnet2.bin.launch \
+		      --cmd "${cuda_cmd} --name ${jobname}" \
+		      --log "${asr_exp}"/train.log \
+		      --ngpu "${pretrain_ngpu}" \
+		      --num_nodes "${pretrain_num_nodes}" \
+		      --init_file_prefix "${asr_exp}"/.dist_init_ \
+		      --multiprocessing_distributed false -- \
+		      ${python} -m espnet2.bin.hubert_train \
+                      --use_preprocessor true \
+		      --normalize null \
+                      --bpemodel none \
+                      --token_type word \
+		      --token_list "${dictdir}/tokens.txt" \
+                      --non_linguistic_symbols none \
+                      --cleaner "${cleaner}" \
+                      --g2p "${g2p}" \
+                      --valid_data_path_and_name_and_type "${_asr_valid_dir}/${_scp},speech,${_type}" \
+                      --valid_data_path_and_name_and_type "${_asr_valid_dir}/text,text,text" \
+                      --valid_shape_file "${asr_stats_dir}/valid/speech_shape" \
+                      --valid_shape_file "${asr_stats_dir}/valid/text_shape.word" \
+                      --resume true \
+                      --fold_length "${_fold_length}" \
+                      --fold_length "${asr_text_fold_length}" \
+                      --output_dir "${asr_exp}" \
+		      --hubert_dict ${dictdir} \
+                      ${_opts}
+	else
+	    log "Skip the training stages"
+	fi
+    done
+fi
+set -x
+if [ ${stop_stage} -ge 6 ]; then
+    ./asr.sh \
+	--stage "${stage}" --stop-stage "${stop_stage}" \
```

Comment:
The data preparation (local/data.sh) is executed again? Does it matter?
It's better not to use "${stage}" here. If someone runs "hubert_asr.sh" from the middle, this would be affected.

Reply:
Make sense! Changed to ”asr_stage=$((stage>6 ? stage : 6))“